clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2727,12 +2724,8 @@ def FuncOp : CIR_Op<"func", [
 
     bool isDeclaration();
 
-    // FIXME: should be shared with GlobalOp extra declaration.
-    bool isDeclarationForLinker() {
-      if (mlir::cir::isAvailableExternallyLinkage(getLinkage()))
-        return true;
-
-      return isDeclaration();
+    bool hasAvailableExternallyLinkage() {
+      return mlir::cir::isAvailableExternallyLinkage(getLinkage());
```

Comment:
Given both hasAvailableExternallyLinkage impls are equal, just make this the default impl in the td file, and delete the one from GlobalOp.

Reply:
I actually tried that. But it fails to compile as tbgen doesn't recognize getLinkage(), and getLinkage() seems to be a getter for OP attribute  "linkage".  Would love to learn how we deal with this situation as I totally agree with you this code dup is ugly.

Reply:
Maybe something along the lines of what we see in `CIR_CallOp` could be doable:
```
    /// Return the callee of this operation
    CallInterfaceCallable getCallableForCallee() {
      return (*this)->getAttrOfType<SymbolRefAttr>("callee");
    }
```
Note that because follow up work will make `mlir::cir::isAvailableExternallyLinkage` be part of the interface as well, perhaps you can address this as part of that change instead.